docs/YOSEMITE.md

```diff
@@ -0,0 +1,103 @@
+# Yosemite
+> This module encapsulates the application business logic, and interacts with the Networking and Storage layers.   
+
+## High-level class diagram
+![Yosemite high level class diagram](images/yosemite_general.png)
+
+## Application state. StoresManagerState
+Business logic is broken down into different subclasses of   `Store`   A `Store` encapsulates the business logic related to one and only one of the domain level concerns (i.e. Notifications, Orders, Shipment…)
+
+Some of these concerns require a valid user session (a user to be logged into the system) while some others don’t require a session.  To handle that, `Store`s are aggregated by a `StoresManager`.
+
+`StoresManager` is a state machine, that manages two states. Those states are implementations of the `StoresManagerState` protocol. Currently, we have two implementations, `AuthenticateState` and `DeauthenticateState`
+
+Each of the implementations of the `StoresManagerState` aggregates a collection of subclasses of `Store` protocol. That’s how we enforce certain concerns (i.e. Orders) to require users to be authenticated.
+
+`StoresManager` takes care of the transition from the authenticated (`AuthenticateState`) to the deauthenticated (`Deauthenticate`) state, while executing the necessary side effects of those transitions (for example, clearing the user data locally stored)
+
+`StoresManager` is a singleton. 
+
+The `AuthenticateState`aggregates a reference to `Dispatcher`. This is an important piece of the system, as it is responsible for executing the actions declared in the different Stores. Therefore we will discuss it in depth later on.
```

Comment:
This piece was a "gotcha!" and I could not figure out why my new dispatched actions weren't working. It's because I forgot to register the new store inside of the services list in the init.

Reply:
Is there a way to make this point stand out more? an ℹ️ emoji or something?